src/app/shared/services/query.service.ts

```diff
@@ -14,11 +14,12 @@ export class QueryService {
 
   constructor(private http: Http) {}
 
-  save(userActions: UserActions, checkedRecords: string[]): Promise<void> {
+  save(userActions: UserActions, checkedRecords: string[], allSelected: boolean): Promise<void> {
```

Comment:
last two params should be optional. because you should either pass `allSelected` or `selectedRecordIds`